project_setup/Skeletonize-existing-repository.md

```diff
@@ -0,0 +1,84 @@
+# Skeletonize an existing repository #
+
+Skeletonizing a repository standardizes our development setup and environment
+as well as enables our [Lineage](https://github.com/cisagov/action-lineage/)
+Github Action to keep it updated and standardized.
+
+## About skeletons ##
+
+Skeleton projects contain [licensing information](LICENSE), as
+well as [pre-commit hooks](https://pre-commit.com) and
+[GitHub Actions](https://github.com/features/actions) configurations
+appropriate for the major languages that we use. This lets us standardize
+[cisagov](https://github.com/cisagov) GitHub projects to a
+[list of cisagov skeleton projects](https://github.com/search?q=org%3Acisagov+skeleton&type=Repositories).
+
+## General overview ##
+
+The general outline of how to add a skeleton to a repository is:
+
+1. Add the skeleton as a remote to the non-skeletonized repository
+2. Pull with `--allow-unrelated-histories`
+3. Fix all the inevitable conflicts (this takes the longest)
+4. Run `pre-commit install`
+5. Fix problems that may arise from `pre-commit` being added
+6. Commit to a branch and submit a PR
+
+## Adding a skeleton as remote ##
+
+### Choose a skeleton ###
+
+First, decide which of the available skeletons fits your existing repository.
+As an example, we'll be using [`skeleton-python-library`](https://github.com/cisagov/skeleton-python-library)
+in this document.
+
+```console
+cd <target_repository>
+git remote add skeleton-parent git@github.com:cisagov/skeleton-python-library.git
+
+# You can verify the remote has been added by
+git remote --verbose
+
+# Create a new branch for this work
+git checkout -b skeletonize
+
+# Pull skeleton's history
+git pull skeleton-parent develop --allow-unrelated-histories
+```
+
+## Fix merge conflicts ##
+
+This merge process will almost certainly fail, resulting in merge conflicts.
+The next step is to fix those conflicts and add the files once the fixes are
+in place.
```

Comment:
You don't just have to fix merge conflicts.  It is important to also look at the _unconflicted_ changes listed in the output of `git status` and verify that you want to include all those changes.

This step is often overlooked because it is rarely needed, but it can save you from merging in destructive upstream changes.

Reply:
I always run over a `git diff --cached` to double check that all the changes make sense before committing.

Reply:
Ah, I usually do `git diff origin/develop`.

Reply:
Ah, yeah I meant just in general and not just when skeletonizing. Yours makes sense for a skeleton compare.

Reply:
I thumbs-upped the initial comment thinking, "Man, I would never have remembered to mention that!" before realizing it was actually written by me. :laughing:

Reply:
I added another section with some info in https://github.com/cisagov/development-guide/pull/32/commits/d062c46acca321119bd4f4998b5a2652948fb345

Reply:
See [here](https://github.com/cisagov/development-guide/pull/32#commitcomment-48188207).